### DISCUSSION TITLE
Install on existing web server like Virtualmin?

### DISCUSSION THREAD

**Discussion by zerovertex** [immense/Remotely#253] [Q&A]
What steps would I take to install on an existing web server like Virtualmin?

I already have it installed and working on a stand alone server but I'd like to move remotely to my shared web server running Virtualmin with Nginx instead of Apache.

**Comment by bitbound** [comment]
I've never run Remotely on shared hosting, but I imagine it would have the same requirements as a VM.

- The .NET process needs to be running and allowed to listen locally for proxied web requests. Default port is 5000, but you can change it with a command line argument (e.g. `--urls http://127.0.0.1:8080`).
- Requests need to be proxied to the port it's using.
- Headers need to be forwarded correctly.
- The websocket/SignalR endpoints need to be upgraded to websocket connections.

You can see the Nginx configuration here:  https://github.com/lucent-sea/Remotely/blob/master/Utilities/Ubuntu_Server_Install.sh 

How that's achieved with Virtualmin, I don't know.  I've never used it.

Hope the above helps, though.

**Reply by houstonbofh** [reply]
If you install in in Linux, you have the nginx config it needs for that virtual host name.  It would be trivial to move everything to another Linux server as long as you have all the needed packages.

**Reply by zerovertex** [reply]
Jared @lucent-sea  - Keep doing awesome work! Thank you!
Lee @houstonbofh - Thanks for the input. I did get the files and db moved over without issue. 
This is probably more an issue with Virtualmin and less a Remotely.one issue. The install script created the Nginx config correctly but the way Virtualmin has Nginx configured when I go to the Remotely.one URL it goes to the first virtual host on the server. I'll post in the Virtualmin forum to see if I can get some more specific assistance with the Nginx configuration.

**Reply by houstonbofh** [reply]
Did you copy over the remotely config file from sites-available and link it properly in sites-enabled?  And does the file have all of the potential domain names? (I had to add the local friendly name) If so, it should work.

**Reply by zerovertex** [reply]
Copied the remotely conf over from the other server. Server is pulling up "domain.co" when I try to go to "remote.domain.co".
Links to the Nginx Conf files, slightly edited.
remotely.domain.co.conf - http://nc.edgartech.co/index.php/s/CFJ4JC9arxHgckK
server.domain.co.conf - http://nc.edgartech.co/index.php/s/yKZCPyHiHobMGDr
Any advice? I'm used to working with Apache and not Nginx.

**Reply by houstonbofh** [reply]
The only obvious thing I see is the server name line.  Mine looks like;

server_name   remotely remotely.domain.co *.remotely.domain.co;

I added the short name as I use it often locally.  I also remove a "\" that you had there. Also I saw above you are using "remote.domain.co" and not "remotely.domain.co" which would pipe you to the default web page.

**Reply by bitbound** [reply]
I've been using Caddy Server lately.  The config file is simply:

```
app.remotely.one {
    reverse_proxy 127.0.0.1:5000
}
```

And that's it.  It also handles Let's Encrypt certificates automatically.

Might be worth checking out if Nginx continues to be a pain.

https://caddyserver.com/

**Reply by zerovertex** [reply]
No dice @houstonbofh. I added remotely to the conf and restarted. 
I've cross posted the configs in the Virtualmin forums, too. https://forum.virtualmin.com/t/running-remotely-one-and-virtualmin-cant-get-nginx-to-load-remotely-one/110227
@lucent-sea I'd like to host more than just remotely on the box and I'd like to not have to use a different port so the goal is get is all running on the same Nginx server.
